sailor/sap_iot/_common.py

```diff
@@ -0,0 +1,59 @@
+import string
+
+from cachetools.func import ttl_cache
+
+from ..utils.config import SailorConfig
+from ..utils.oauth_wrapper import get_oauth_client
+
+
+class _CustomFormatter(string.Formatter):
+    # Since the URL returned by the extension service contains the same format string key twice
+    # this formatter can be used to fill those sequentially from an iterable
+    def get_value(self, key, args, kwargs):
+        return kwargs['replace'].pop(0)
```

Comment:
I have not much experience with string formatting. this seems hacky, but simple enough. kudos :)